docs/design.md

```diff
@@ -0,0 +1,131 @@
+
+It's main features are:
+* Trackers: simple panels that track amongst others: mood, number of cigarettes smoked, step count, thought records.
+* Daily planner: overview of planned actions
+* Messaging
+ 
+#### Sensor data collection app (out of scope)
+A separate app for collecting sensor data that is needed to estimate physical capacity. 
+* The Niceday app itself cannot gather this data.
+* At the start of an activity the sensor data collector is triggered to start recording sensor data.
+* After an activity or upon Wifi connection the data is pushed to the PerfectFit sensor data REST API.
+* Will be based on an existing app and developed outside of scope of PerfectFit
+virtual coach system.
+--- 
+
+### Niceday components
+#### Niceday server (already developed)
+* Backend for the Niceday app.
+* Data storage: 
+  - user profile data, 
+  - app-interaction data
+  - messages
+* Handles authentication
+
+--- 
+### PerfectFit virtual coach system components
+We cannot change much in the NiceDay app, new PerfectFit features will be implemented 
+through the virtual coach that interacts with the user through the messaging function of the 
+NiceDay app. We will make use of Niceday trackers to gather data, 
+and make use of the Niceday daily planner to plan activities.
+
+
+### Sensor data REST API
+A Sensor REST API that allows for the sensor data collection app to send data to PerfectFit system.
+* Writes data directly to the database using the ORM
+
+### Rasa agent
+The conversational agent developed in the [Rasa](https://rasa.com/) framework.
+* Given an input message of the user, perform the appropriate action.
+* Actions can range from a simple text message response to something more complex, such as:
+  - using an algorithm (through importing the library) 
+  - querying the perfectfit database using ORM
+  - calling the niceday-api for information from niceday-server
+
+### Rasa agent REST API
+Allows for other components to interact with the agent.
+
+### niceday-broker
+Sends back and forth messages from a niceday user between the Niceday server and the rasa agent.
+* Depends on the javascript Niceday client [goalie-js](https://github.com/senseobservationsystems/goalie-js).
+* Listens to incoming messages, upon an incoming message it sends a request to the rasa agent REST API
+* It forwards any text response from rasa back to the user through Niceday
+
+### niceday-api
+Allows control over certain functionalities in the niceday app.
+It is a node.js REST API that wraps functions that we need from the javascript Niceday client 
+[goalie-js](https://github.com/senseobservationsystems/goalie-js).
+* Request data from the niceday server
+* Read from and manage Niceday (custom) trackers
+* Interact with Niceday daily planner
+
+### PostgreSQL database
+Stores PerfectFit-specific data, 
+* Stores:
+  - Data about the user (name, age, stage of patient journey, preferences)
+  - Sensor data
```

Comment:
Will this need to be wiped periodically to avoid filling up the db? If so, will it be the algorithm's job to indicate when it is finished with it? What if there is more than one algorithm that wants to process the same sensor data? Perhaps some sort of expiry-date should also be written to the db. (Sorry, probably too technical for this high level design discussion)

Reply:
Good one, I added a section on 'workers' that might do the processing.

And indeed an expiry-date is useful, especially for sensor data. I added a bit about that as well.